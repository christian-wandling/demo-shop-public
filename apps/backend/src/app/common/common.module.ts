import { Module } from '@nestjs/common';
import { PrismaService } from './services/prisma.service';
import { MonitoringService } from './services/monitoring.service';

@Module({
  providers: [PrismaService, MonitoringService],
  exports: [PrismaService, MonitoringService],
})
export class CommonModule {}
