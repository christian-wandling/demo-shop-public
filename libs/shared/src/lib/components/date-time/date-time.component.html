<time [dateTime]="dateTime()">{{ dateTime() | date : 'MMM dd, YYYY' : 'GMT' }}</time>
