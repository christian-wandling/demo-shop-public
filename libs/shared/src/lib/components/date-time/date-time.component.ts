import { ChangeDetectionStrategy, Component, input, InputSignal } from '@angular/core';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'lib-date-time',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './date-time.component.html',
  styleUrl: './date-time.component.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class DateTimeComponent {
  /**
   * Date to display
   */
  readonly dateTime: InputSignal<Date | string> = input.required<Date | string>();
}
