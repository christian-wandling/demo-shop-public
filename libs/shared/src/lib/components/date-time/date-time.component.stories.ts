import { Meta, moduleMetadata, StoryObj } from '@storybook/angular';
import { DateTimeComponent } from './date-time.component';
import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { CommonModule } from '@angular/common';

const meta: Meta<DateTimeComponent> = {
  component: DateTimeComponent,
  title: 'Shared/DateTimeComponent',
  decorators: [
    moduleMetadata({
      imports: [CommonModule],
    }),
  ],
};
export default meta;
type Story = StoryObj<DateTimeComponent>;

export const WithDate: Story = {
  args: {
    dateTime: new Date('2025-03-16T12:00:00'), // Example with specific date
  },
  argTypes: {
    dateTime: {
      control: {
        type: 'date',
      },
    },
  },
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const date = args.dateTime.toLocaleString('en-US', {
      month: 'short',
      day: '2-digit',
      year: 'numeric',
    });

    expect(canvas.getByText(date)).toBeTruthy();
  },
};

export const WithISOString: Story = {
  args: {
    dateTime: '2025-03-16T12:00:00Z',
  },
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const date = new Date(args.dateTime).toLocaleString('en-US', {
      month: 'short',
      day: '2-digit',
      year: 'numeric',
    });

    expect(canvas.getByText(date)).toBeTruthy();
  },
};
