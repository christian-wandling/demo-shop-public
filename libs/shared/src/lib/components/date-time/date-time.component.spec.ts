import { ComponentFixture, TestBed } from '@angular/core/testing';
import { DateTimeComponent } from './date-time.component';
import { By } from '@angular/platform-browser';
import { CommonModule } from '@angular/common';

describe('DateTimeComponent', () => {
  let component: DateTimeComponent;
  let fixture: ComponentFixture<DateTimeComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [DateTimeComponent, CommonModule],
    }).compileComponents();

    fixture = TestBed.createComponent(DateTimeComponent);
    component = fixture.componentInstance;

    // Set a fixed date for consistent testing
    const testDate = new Date('2025-03-16T12:00:00+00:00');
    fixture.componentRef.setInput('dateTime', testDate);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should correctly display the formatted date', () => {
    const timeElement = fixture.debugElement.query(By.css('time')).nativeElement;
    expect(timeElement.textContent.trim()).toBe('Mar 16, 2025');
  });

  it('should handle string date input', () => {
    const stringDate = '2025-01-01T00:00:00Z';
    fixture.componentRef.setInput('dateTime', stringDate);
    fixture.detectChanges();

    const timeElement = fixture.debugElement.query(By.css('time')).nativeElement;
    expect(timeElement.textContent.trim()).toBe('Jan 01, 2025');
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('with different date formats', () => {
    it('should handle future dates', () => {
      const futureDate = new Date('2026-12-25T00:00:00Z');
      fixture.componentRef.setInput('dateTime', futureDate);
      fixture.detectChanges();

      const timeElement = fixture.debugElement.query(By.css('time')).nativeElement;
      expect(timeElement.textContent.trim()).toBe('Dec 25, 2026');
      expect(fixture).toMatchSnapshot('future-date');
    });

    it('should handle past dates', () => {
      const pastDate = new Date('2020-04-01T00:00:00Z');
      fixture.componentRef.setInput('dateTime', pastDate);
      fixture.detectChanges();

      const timeElement = fixture.debugElement.query(By.css('time')).nativeElement;
      expect(timeElement.textContent.trim()).toBe('Apr 01, 2020');
      expect(fixture).toMatchSnapshot('past-date');
    });
  });
});
