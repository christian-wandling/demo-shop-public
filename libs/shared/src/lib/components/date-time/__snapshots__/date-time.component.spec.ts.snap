// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeComponent should match snapshot 1`] = `
<lib-date-time
  dateTime={[Function Function]}
  pattern={[Function Function]}
  timezone={[Function Function]}
>
  <time
    datetime="Mar 16, 2025"
  >
     Mar 16, 2025
  </time>
</lib-date-time>
`;

exports[`DateTimeComponent with different date formats should handle future dates: future-date 1`] = `
<lib-date-time
  dateTime={[Function Function]}
  pattern={[Function Function]}
  timezone={[Function Function]}
>
  <time
    datetime="Dec 25, 2026"
  >
     Dec 25, 2026
  </time>
</lib-date-time>
`;

exports[`DateTimeComponent with different date formats should handle past dates: past-date 1`] = `
<lib-date-time
  dateTime={[Function Function]}
  pattern={[Function Function]}
  timezone={[Function Function]}
>
  <time
    datetime="Apr 01, 2020"
  >
     Apr 01, 2020
  </time>
</lib-date-time>
`;
